### DISCUSSION TITLE
Prevent Inertia from replacing the url

### DISCUSSION THREAD

**Discussion by ltraccard-dynadmic** [inertiajs/inertia#779] [Ideas]
Hi, 

I am trying to implement a page where I load a resource with an infinite scroll. Everything is working fine, except for one little thing.

Visiting /users will render the first 25 users, scrolling down will load page 2, the received dataset will be added to the initial dataset and the url will be rewritten to /users?page=2, and so on.

I do not offer any other way to navigate trough the dataset, so the problem occurs when you access a page with a page parameter.

Visiting /users?page=2 will render the users from 26 to 50, it's now impossible to access the first 25 users with infinite scrolling.

I found a fix but the url is now flashing. 

```javascript 
Inertia.on('navigate', (event) => {
    const url = hrefToUrl(event.detail.page.url);
    url.searchParams.delete('cursor');
    window.history.replaceState(event.detail.page, '', url.toString());
})
```

I will be great if we could prevent Inertia from replacing the state, or some given parameters. I can make a PR if you think it could be a good addition to Inertia.

**Comment by ragulka** [comment]
I'd love to have this. It looks like at the moment when `setPage` is called, it will _always_ either replace or push the history state. Perhaps it would be possible to add a 3rd option which would simply set the page, but wouldn't touch the history at all?

https://github.com/inertiajs/inertia/blob/2595fa34d3f4e1502aa2c980f626c8ca51e7d97a/packages/core/src/router.ts#L457

**Comment by knightspore** [comment]
I found a solution where adding this router hook *outside* of `createInertiaApp` seems to work quite well, without any flashing.

**Reply by ignacio-dev** [reply]
Also, Inertia v2 is nearly out of beta, and it offers infinite scroll without url replacement ☺️

**Reply by aphrxia** [reply]
@ignacio-dev any information on the schedule for the v2 stable release? TIA

**Reply by brendanpetty** [reply]
> Also, Inertia v2 is nearly out of beta, and it offers infinite scroll without url replacement ☺️

I was hoping that to be true, but it doesn't seem to work (and there's no documentation about this). Url is replaced just as normal. Am I missing something?

**Reply by brendanpetty** [reply]
The latest release (including https://github.com/inertiajs/inertia/pull/2135) appears to have fixed this for me... I can use `preserveUrl: true` and everything works as I hoped for. @aphrxia @ragulka
